core/tests/test_utils.py

```diff
@@ -1076,6 +1076,10 @@ def save_new_valid_exploration(
             language_code=language_code)
         exploration.states[exploration.init_state_name].update_interaction_id(
             interaction_id)
+
+        if interaction_id == 'EndExploration':
+            exploration.states[exploration.init_state_name].update_interaction_default_outcome(None) # pylint: disable=line-too-long
```

Comment:
You don't need this, right? Can't you just set an `end_state_name` value and the 'if' block below would populate everything correctly?

Reply:
Actually, the EndExploration interaction has `is_terminal=true` by default so it wouldn't make a difference to that attribute's value even if we attach another EndExploration interaction to the original one.
So, even if we were to set an `end_state_name`, the interaction would still be considered as terminal unless we explicitly change the `is_terminal` value, which is what I've done right now.  Also, it doesn't really make sense if two EndExploration interactions are present consecutively.